src/main/protocol-handler/lens-protocol-router-main/lens-protocol-router-main.ts

```diff
@@ -40,11 +39,15 @@ function checkHost<Query>(url: URLParse<Query>): boolean {
 interface Dependencies {
   extensionLoader: ExtensionLoader;
   extensionsStore: ExtensionsStore;
+  showApplicationWindow: () => Promise<void>;
 }
 
 export class LensProtocolRouterMain extends proto.LensProtocolRouter {
   private missingExtensionHandlers: FallbackHandler[] = [];
 
+  /**
+   * @deprecated This is used to solve out-of-place temporal dependency. Remove, and solve otherwise.
```

Comment:
I don't think this should be marked as deprecated if it is merely considered bad. The needs to be a replacement immediately usable when something is deprecated.

Reply:
Changed to TODO.